lib/connections/service.go

```diff
@@ -396,6 +407,11 @@ func (s *service) connect(ctx context.Context) {
 			dialTargets := make([]dialTarget, 0)
 
 			for _, addr := range addrs {
+				select {
+				case <-ctx.Done():
+					return
+				default:
+				}
 				// Use a special key that is more than just the address, as you might have two devices connected to the same relay
```

Comment:
I don't think this loop does anything lengthy, the dial is the thing that takes time.

Reply:
I simply didn't check what it does exactly, I just added it based on the loop internals being lengthy (as in lines of code lengthy) and thus wanting to be safe.